docs/software/apptainer.md

```diff
@@ -0,0 +1,233 @@
+
+=== "Engaging"
+
+     Here is an example job script to run a python program with a GPU and data files saved in `/nobakcup1` or `/pool001` directories,
+     ```
+     #!/bin/bash                      
+     #SBATCH -t 01:30:00         # walltime = 1 hours and 30 minutes
+     #SBATCH -N 1                # one node
+     #SBATCH -n 2                # two CPU cores
+     #SBATCH --gres=gpu:1        # one GPU
+     #SBATCH -p sched_mit_psfc_gpu_r8     # a partition with Rocky 8 nodes
+
+     module load apptainer/1.1.7-x86_64 squashfuse/0.1.104-x86_64   # load modules
+     singularity exec --nv -B /nobakcup1,/pool001 my-image.sif python my-code.py   # Run the program
+     ```
+
+=== "OpenMind"
+
+     Here is an example job script to run a Python program with a GPU and data files saved in `/om` or `/om2` directories,
+     ```
+     #!/bin/bash                      
+     #SBATCH -t 01:30:00             # walltime = 1 hours and 30 minutes
+     #SBATCH -N 1                    # one node
+     #SBATCH -n 2                    # two CPU cores
+     #SBATCH --gres=gpu:1            # one GPU
+     #SBATCH --constraint=rocky8     # nodes with Rocky 8 OS
+
+     module load openmind8/apptainer/1.1.7        # load an apptainer module
+     singularity exec --nv -B /om,om2 my-image.sif python my-code.py  # Run the program
+     ```
+
+## Build Singularity images
+
+In the previous section, it is assumed that all needed packages have been installed in the image. If some needed packages do not exist in the image, users need to build a new image. 
+
+To save work for the building process, search for an image providing the right OS and necessary dependencies to support your target application, then use it as a base image and build your target application on top of it. 
+
+The following is an example of building Python packages such as Pytorch and Pandas in a container image. 
+
+First, download a Docker image that provides the Ubuntu OS and have Python and PyTorch installed already,
+```
+singularity build --sandbox my-image  docker://bitnami/pytorch:latest
+```
+
+The command `build` here does not build anything yet, but just downloads the image and converts it to a new format. The flag `--sandbox` tells `build` to convert the image to the Sandbox format, which is convenient for installing packages interactively. 
+
+Log in to the container shell, then you can install system packages using `apt-get` as is on an Ubuntu machine and build Python packages using `pip install`, taking Pandas for example, 
+```
+$ singularity shell --writable my-image
+Apptainer> apt-get update
+Apptainer> pip install pandas
+Apptainer> python 
+Python 3.8.17 (default, Jun 16 2023, 21:48:21) 
+[GCC 10.2.1 20210110] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import pandas as pd
+```
+
+The flag `--writable` is to enable the write permission to modify files in the container. 
+
+??? note
```

Comment:
Remove this note, if we add the note above that building containers on Engaging is not supported currently we don't need this note here.